indexer/routing.cpp

```diff
@@ -0,0 +1,58 @@
+#include "indexer/routing.hpp"
+
+namespace
+{
+char const kNo[] = "No";
+char const kOnly[] = "Only";
+}  // namespace
+
+namespace routing
+{
+// static
+uint32_t const Restriction::kInvalidFeatureId = numeric_limits<uint32_t>::max();
+
+bool Restriction::IsValid() const
+{
+  return !m_featureIds.empty() && find(begin(m_featureIds), end(m_featureIds), kInvalidFeatureId)
+      == end(m_featureIds);
+}
+
+bool Restriction::operator==(Restriction const & restriction) const
+{
+  return m_featureIds == restriction.m_featureIds && m_type == restriction.m_type;
+}
+
+bool Restriction::operator<(Restriction const & restriction) const
+{
+  if (m_type != restriction.m_type)
+    return m_type < restriction.m_type;
+  return m_featureIds < restriction.m_featureIds;
+}
+
+string ToString(Restriction::Type const & type)
```

Comment:
If this function is used in DebugPrint() only, please, get rid of this function.